### PR TITLE
Feature: Implemented openQASM serialization of ECR gate

### PR DESCRIPTION
**Pull request by synthia2080**
*Issue #, if available:*
#329 

*Description of changes:*
Implemented openQASM serialization of ECR gate

*Testing done:*
Ran linters and unit-tests. Practice code was ran but with error of device incompatibility.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by wrasmuss — APPROVED** [review]
Looks Good!

**Comment by synthia2080** [comment]
> Looks Good!

Thank you!!

File: src/braket/circuits/gates.py [hunk 1]
```diff
@@ -1514,6 +1514,13 @@ def adjoint(self) -> List[Gate]:
     def _to_jaqcd(self, target: QubitSet):
         return ir.ECR.construct(targets=[target[0], target[1]])
 
+    def _to_openqasm(
+        self, target: QubitSet, serialization_properties: OpenQASMSerializationProperties, **kwargs
+    ):
+        target_qubit0 = serialization_properties.format_target(int(target[0]))
+        target_qubit1 = serialization_properties.format_target(int(target[1]))
+        return f"ecr {target_qubit0}, {target_qubit1};"
```
Comment (ajberdy) [inline src/braket/circuits/gates.py:1522]: nit: I would add a `_` before the index
```suggestion
        return f"ecr {target_qubit_0}, {target_qubit_1};"
```

Comment (ajberdy) [inline src/braket/circuits/gates.py:1522]: nit: I would add a `_` before the index
```suggestion
        target_qubit_0 = serialization_properties.format_target(int(target[0]))
        target_qubit_1 = serialization_properties.format_target(int(target[1]))
        return f"ecr {target_qubit_0}, {target_qubit_1};"
```

**Pull request merged**